File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
// space: o(1)
// time : o(n)
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int profit = 0;
        
        for(int i=1;i<prices.size();i++){
            if(prices[i] > prices[i-1]){
                profit += prices[i] - prices[i-1];
            }
            
        }
        return profit;
        
    }
};

/*

int main() {
    Solution solution;
    
    // Create a vector representing stock prices
    vector<int> prices = {7, 1, 5, 3, 6, 4};
    
    // Call the maxProfit function to calculate the maximum profit
    int result = solution.maxProfit(prices);
    
    // Print the result
    cout << "Maximum profit: " << result << endl;
    
    return 0;
}
*/

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
## space: o(1)
## time : o(n)
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit =0
        
        ## interate from the first value in arrays beacuse we have to compare the previous value for the current ith , hence 0th value dosen't have any  previous value , start the iteration from 1st place
        for i in range(1 , len(prices)):
            ## the above conditional statement inidicates profit
            if(prices[i] > prices[i-1]):
                profit += (prices[i] - prices[i-1])
                
        return profit        
        
        
# Test the maxProfit function
#solution = Solution()

# Create a list representing the stock prices
#prices = [7, 1, 5, 3, 6, 4]

# Call the maxProfit function to calculate the maximum profit
#result = solution.maxProfit(prices)

# Print the result
#print("Maximum profit:", result)#        